src/Map/src/Marker.php

```diff
@@ -38,4 +49,28 @@ public function toArray(): array
             'extra' => (object) $this->extra,
         ];
     }
+
+    /**
+     * @param array{
+     *     position: array{lat: float, lng: float},
+     *     title: string|null,
+     *     infoWindow: array<string, mixed>|null,
+     *     extra: object,
+     * } $marker
+     *
+     * @internal
+     */
+    public static function fromArray(array $marker): self
+    {
+        if (!isset($marker['position'])) {
+            throw new InvalidArgumentException('The "position" parameter is required.');
+        }
+        $marker['position'] = Point::fromArray($marker['position']);
```

Comment:
```suggestion
        $marker['position'] = Point::fromArray($marker['position'] ?? throw new InvalidArgumentException('The "position" parameter is required.'));
```

Reply:
I'm not fan of exceptions inside a parenthesis to be honest..

Reply:
I prefer not to change this one for now, but thanks for all the small optimizations suggested.. in the end i know this adds up 😃